File: src/test/java/com/ject/studytrip/auth/helper/AuthCookieTestHelper.java
package com.ject.studytrip.auth.helper;

import jakarta.servlet.http.Cookie;
import org.springframework.stereotype.Component;

@Component
public class AuthCookieTestHelper {

    public Cookie createKakaoSignupProfileCookie(String name, String signupKey) {
        Cookie cookie = new Cookie(name, signupKey);
        cookie.setHttpOnly(true);
        cookie.setSecure(true);
        cookie.setPath("/");
        return cookie;
    }

    public Cookie createRefreshTokenCookie(String name, String refreshToken) {
        Cookie cookie = new Cookie(name, refreshToken);
        cookie.setHttpOnly(true);
        cookie.setSecure(true);
        cookie.setPath("/");
        return cookie;
    }
}

File: src/test/kotlin/com/ject/studytrip/auth/helper/AuthCookieTestHelper.kt
package com.ject.studytrip.auth.helper

import jakarta.servlet.http.Cookie
import org.springframework.stereotype.Component

@Component
class AuthCookieTestHelper {
    fun createKakaoSignupProfileCookie(
        name: String,
        signupKey: String,
    ): Cookie =
        Cookie(name, signupKey).apply {
            isHttpOnly = true
            secure = true
            path = "/"
        }

    fun createRefreshTokenCookie(
        name: String,
        refreshToken: String,
    ): Cookie =
        Cookie(name, refreshToken).apply {
            isHttpOnly = true
            secure = true
            path = "/"
        }
}
